sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/object.scala

```diff
@@ -41,7 +41,20 @@ object CatalystSerde {
   }
 
   def generateObjAttr[T : Encoder]: Attribute = {
-    AttributeReference("obj", encoderFor[T].deserializer.dataType, nullable = false)()
+    val deserializer = encoderFor[T].deserializer
+    val dataType = deserializer.dataType
+    val nullable = if (deserializer.childrenResolved) {
```

Comment:
The point is, `encoderFor` always returns an unresolved `ExpressionEncoder`. As after the line 44 `val deserializer = encoderFor[T].deserializer`, you never try to resolve this encoder, how does it become resolved?